### ISSUE TITLE
thread_local doesn't list the same caveats as __declspec(thread) regarding dynamically-loaded DLLs

### ISSUE THREAD

**Issue by ericgrossNI** [MicrosoftDocs/cpp-docs#1921]
__declspec(thread) lists some caveats reagarding the behavior of thread-local variable initialization when used in a DLL. Specifically, that threads existing before the DLL was loaded do not get their initialization function/CTOR run. This appears to apply to the C++11 thread_local storage classs as well.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 66737d90-dbf1-88c9-3fa1-dcb00c335467
* Version Independent ID: 7292bb09-ee1a-1ae6-1985-631cdf24a27d
* Content: [Storage classes (C++)](https://docs.microsoft.com/en-us/cpp/cpp/storage-classes-cpp?view=vs-2019#feedback)
* Content Source: [docs/cpp/storage-classes-cpp.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/cpp/storage-classes-cpp.md)
* Product: **visual-cpp**
* Technology: **cpp-language**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@ericgrossNI 
That's true, and the explanatory link is not immediately apparent. The behavior is changing in 16.5 previews, and the TLS initialization content is due for an update to match soon. I'll foreground it here as well.
